internal/crypto/password.go

```diff
@@ -65,9 +72,94 @@ type Argon2HashInput struct {
 	rawHash []byte
 }
 
+type FirebaseScryptHashInput struct {
+	alg           string
+	v             string
+	memory        uint64
+	rounds        uint64
+	threads       uint64
+	saltSeparator []byte
+	signerKey     []byte
+	salt          []byte
+	rawHash       []byte
+}
+
+func ParseFirebaseScryptHash(hash string) (*FirebaseScryptHashInput, error) {
+	submatch := scryptHashRegexp.FindStringSubmatchIndex(hash)
+	if submatch == nil {
+		return nil, errors.New("crypto: incorrect scrypt hash format")
+	}
+
+	alg := string(scryptHashRegexp.ExpandString(nil, "$alg", hash, submatch))
+	v := string(scryptHashRegexp.ExpandString(nil, "$v", hash, submatch))
+	n := string(scryptHashRegexp.ExpandString(nil, "$n", hash, submatch))
+	r := string(scryptHashRegexp.ExpandString(nil, "$r", hash, submatch))
+	p := string(scryptHashRegexp.ExpandString(nil, "$p", hash, submatch))
+	ss := string(scryptHashRegexp.ExpandString(nil, "$ss", hash, submatch))
+	sk := string(scryptHashRegexp.ExpandString(nil, "$sk", hash, submatch))
+	saltB64 := string(scryptHashRegexp.ExpandString(nil, "$salt", hash, submatch))
+	hashB64 := string(scryptHashRegexp.ExpandString(nil, "$hash", hash, submatch))
+
+	if alg != "fbscrypt" {
+		return nil, fmt.Errorf("crypto: Firebase scrypt hash uses unsupported algorithm %q only fbscrypt supported", alg)
+	}
+	if v != "1" {
```

Comment:
To allow for extensibility, though unlikely, in case we want to move away from the existing set of params